sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/ExecutorRequirement.java

```diff
@@ -55,20 +63,23 @@ private static ExecutorRequirement getExistingExecutorRequirement(ExecutorInfo e
         }
     }
 
-    private ExecutorRequirement(ExecutorInfo executorInfo)
-            throws InvalidRequirementException {
+    private ExecutorRequirement(
+            ExecutorInfo executorInfo,
+            Collection<ResourceRequirement> resourceRequirements) throws InvalidRequirementException {
         validateExecutorInfo(executorInfo);
         this.executorInfo = executorInfo;
+        this.resourceRequirements = resourceRequirements != null ? resourceRequirements :
```

Comment:
This seems a little strange.  So sometimes you're supposed to pass in all `ResourceRequirement`s.  Other times we derive them from the `ExecutorInfo`.  This seems to be a side effect of passing in volume related `ResourceRequirement`s sometimes.  It's a little funny that when we pass in requirements its always ONLY volume requirements.  Why don't we do one or th other.  Either all the `ResourceRequirement`s are derived from the `ExecutorInfo`, or none are.  If none are, why pass in an `ExecutorInfo`? Why couldn't the ExecutorInfo have all the appropriate volume information on it already and have it extracted here?